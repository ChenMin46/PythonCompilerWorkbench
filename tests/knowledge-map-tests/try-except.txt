raise
raise a
raise a  ,  b
raise a  ,b,  c

try:
    x = d[name]
except KeyError:
    print 'not found'
    raise NotImplementedError

try:
    x = 5
except:
    x = 10
