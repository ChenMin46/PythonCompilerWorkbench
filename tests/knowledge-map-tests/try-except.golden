raise
raise a
raise a  ,  b
raise a  ,b,  c

try:
    x = d[name]
except KeyError:
    print 'not found'
    raise NotImplementedError

try:
    x = 5
except:
    x = 10
===
Module
  body: [
    Raise >> 'raise'
      type: None
      inst: None
      tback: None
    Raise >> 'raise a'
      type: Name >> 'a'
        id: 'a'
        ctx: Load
      inst: None
      tback: None
    Raise >> 'raise a  ,  b'
      type: Name >> 'a'
        id: 'a'
        ctx: Load
      inst: Name >> 'b'
        id: 'b'
        ctx: Load
      tback: None
    Raise >> 'raise a  ,b,  c'
      type: Name >> 'a'
        id: 'a'
        ctx: Load
      inst: Name >> 'b'
        id: 'b'
        ctx: Load
      tback: Name >> 'c'
        id: 'c'
        ctx: Load
    TryExcept >> 'try'
      body: [
        Assign >> 'x = d[name]'
          targets: [
            Name >> 'x'
              id: 'x'
              ctx: Store
          ]
          value: Subscript >> 'd[name]'
            value: Name >> 'd'
              id: 'd'
              ctx: Load
            slice: Index >> 'name'
              value: Name >> 'name'
                id: 'name'
                ctx: Load
            ctx: Load
      ]
      handlers: [
        ExceptHandler >> 'except KeyError'
          type: Name >> 'KeyError'
            id: 'KeyError'
            ctx: Load
          name: None
          body: [
            Print >> "print 'not found'"
              dest: None
              values: [
                Str >> "'not found'"
                  s: 'not found'
              ]
              nl: True
            Raise >> 'raise NotImplementedError'
              type: Name >> 'NotImplementedError'
                id: 'NotImplementedError'
                ctx: Load
              inst: None
              tback: None
          ]
      ]
      orelse: [
      ]
    TryExcept >> 'try'
      body: [
        Assign >> 'x = 5'
          targets: [
            Name >> 'x'
              id: 'x'
              ctx: Store
          ]
          value: Num >> '5'
            n: 5
      ]
      handlers: [
        ExceptHandler >> 'except'
          type: None
          name: None
          body: [
            Assign >> 'x = 10'
              targets: [
                Name >> 'x'
                  id: 'x'
                  ctx: Store
              ]
              value: Num >> '10'
                n: 10
          ]
      ]
      orelse: [
      ]
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "id": "tid_1", 
              "name": "token", 
              "value": "raise"
            }
          ], 
          "id": "id_0", 
          "name": "Raise"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_2", 
              "name": "token", 
              "value": "raise"
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "a"
            }
          ], 
          "id": "id_0", 
          "name": "Raise"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_3", 
              "name": "token", 
              "value": "raise"
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "a"
            }, 
            "  ", 
            {
              "id": "tid_4", 
              "name": "token", 
              "value": ","
            }, 
            "  ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "b"
            }
          ], 
          "id": "id_0", 
          "name": "Raise"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_5", 
              "name": "token", 
              "value": "raise"
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "a"
            }, 
            "  ", 
            {
              "id": "tid_6", 
              "name": "token", 
              "value": ","
            }, 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "b"
            }, 
            {
              "id": "tid_7", 
              "name": "token", 
              "value": ","
            }, 
            "  ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "c"
            }
          ], 
          "id": "id_0", 
          "name": "Raise"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_8", 
              "name": "token", 
              "value": "try"
            }
          ], 
          "id": "id_0", 
          "name": "TryExcept"
        }, 
        {
          "id": "tid_9", 
          "name": "token", 
          "value": ":"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_10", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "d"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Name", 
                      "value": "name"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "Index"
                }, 
                "]"
              ], 
              "id": "id_0", 
              "name": "Subscript"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_11", 
              "name": "token", 
              "value": "except"
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "KeyError"
            }
          ], 
          "id": "id_0", 
          "name": "ExceptHandler"
        }, 
        {
          "id": "tid_12", 
          "name": "token", 
          "value": ":"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "tid_13", 
              "name": "token", 
              "value": "print"
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Str", 
              "value": "'not found'"
            }
          ], 
          "id": "id_0", 
          "name": "Print"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "tid_14", 
              "name": "token", 
              "value": "raise"
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "NotImplementedError"
            }
          ], 
          "id": "id_0", 
          "name": "Raise"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "id": "tid_15", 
              "name": "token", 
              "value": "try"
            }
          ], 
          "id": "id_0", 
          "name": "TryExcept"
        }, 
        {
          "id": "tid_16", 
          "name": "token", 
          "value": ":"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_17", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "5"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_18", 
              "name": "token", 
              "value": "except"
            }
          ], 
          "id": "id_0", 
          "name": "ExceptHandler"
        }, 
        {
          "id": "tid_19", 
          "name": "token", 
          "value": ":"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_20", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "10"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
